sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/AnalyticsTransmissionTarget.java

```diff
@@ -61,6 +72,34 @@
         mPropertyConfigurator = new PropertyConfigurator(channel, this);
     }
 
+    /**
+     * Add an authentication provider to associate logs with user identifiers.
+     *
+     * @param authenticationProvider The authentication provider.
+     */
+    public static synchronized void addAuthenticationProvider(AuthenticationProvider authenticationProvider) {
+
+        /* Validate input. */
+        if (authenticationProvider.getType() == null) {
```

Comment:
For my understanding, what does it mean for the type to be `null`? Why not just check if the object is `null` here?

Reply:
The object is user input so it can be null by mistake.

Reply:
Oh got it. I was for some reason thinking that getType was like `typeof`. Either way, null check should be first (and now is)

Reply:
Also why do we use type fields instead of inheritance? Not that we need to change it at this point

Reply:
There is no code specific to type, it will only be used later to have 1 provider per type allowed.